[ONNX] Update Split operator in unsafe_chunk() method.

fatcat-z opened the pull request:
Update unsafe_chunk() method for opset 13 Split.

BowenBao committed:
[ONNX] Use parameter values in onnx shape inference (#49706)

Adds an additional run of onnx shape inference after constant folding, since initializer may have changed and affected shape inference.

BowenBao committed:
[ONNX] Support opset13 Squeeze and Unsqueeze (#50150)

In opset 13, squeeze/unsqueeze is updated to take axes as input, instead of attribute.

fatcat-z committed:
def unsafe_chunk() support and test in ops13.

fatcat-z committed:
def unsafe_chunk() support and test in ops13.

fatcat-z committed:
Use _unsqueeze_helper insteadof Unsqueeze operator

fatcat-z committed:
Use _unsqueeze_helper insteadof Unsqueeze operator

hwangdeyu committed:
[ONNX] Update Reducesum operator for opset 13 (#50532)

* udpate symbolic for squeeze/unsqueeze

* update c++ unsqueeze/squeeze creation

* clang format

* enable tests

* clang format

* remove prints

* remove magic number

* add helper function

* fix build issue

* update opset9 symbolic with helper function

* fix utility test

* fix prim_fallthrough opset skip

* enable reducesum opset 13

* enable embedding_bag which contain reducesum op

* add ReduceSum helper

* remove block_listed_operators

* remove local test code

* remove embedding_bag() in opset13 file

* remove unuse import

Co-authored-by: BowenBao <bowbao@microsoft.com>
Co-authored-by: hwangdeyu <deyhuang@qq.com>

hwangdeyu committed:
[ONNX] Add binary_cross_entropy_with_logits op to ONNX opset version 12 (#49675)

Fixes #{#47997}
Exporting the operator binary_cross_entropy_with_logits to ONNX opset version 12.

hwangdeyu committed:
[ONNX] Add logical_and, logical_or, logical_xor torch op support in pytorch exporter (#50570)

Fixes #{}
Add logical_and, logical_or, logical_xor torch op support in pytorch exporter.

jiafatom committed:
[ONNX] Enable _jit_pass_onnx_fold_if only when dynamic_axes is None (#50582)

Fixing pytorch/vision#3251 (PR #49410 triggers the torch vision test build failure, on three tests test_faster_rcnn, test_mask_rcnn, test_keypoint_rcnn. )

The offending PR is fine on pytorch UT, because the torchvision and pytorch test has a gap when we merge them - we are using different test API on two sides, therefore causing some discrepancy.

This PR bridge the gap for the above three tests, and disable _jit_pass_onnx_fold_if pass until it gets fixed.
Allow _jit_pass_onnx_fold_if only when dynamic_axes is None.

fatcat-z committed:
Cast the splits into long.

fatcat-z committed:
Cast the splits into long.

BowenBao committed:
[ONNX] Support gelu for fp16 export (#50487)

Need to replace dtype of export created scalars from float to double. (In torch implicit conversion logic, python numbers are double)

Test case skipped in CI due to that current CI job env does not have CUDA support.

KsenijaS commented on test/onnx/test_pytorch_onnx_onnxruntime.py:
```diff
@@ -6200,6 +6200,26 @@ def forward(self, images, features):
+    @disableScriptTest()
+    def test_unsafe_chunk(self):
+        class ChunkModel(torch.nn.Module):
+            def __init__(self):
```
why is init is needed here?

fatcat-z replied:
Good catch. It is not necessary.

BowenBao commented on torch/onnx/symbolic_opset13.py:
```diff
@@ -142,3 +142,26 @@ def reduce_dim(g, self, dim, keepdim, dtype):
+                    g.op("Constant", value_t=torch.tensor(1, dtype=torch.long)),
+                    axis_i=dim, keepdims_i=0)
+
+    size = sym_help._get_tensor_dim_size(self, dim)
```
could `size` potentially be a symbol?

neginraoof replied:
This part could be implemented more dynamically, but we were looking at the conditional logic in line 160. The dynamic implementation of this part was more complex in this case.

fatcat-z committed:
Change the test to a fixed dimension.

fatcat-z committed:
[ONNX] Replace optional parameters of Resize with placeholder for ops13. (#50574)

* Replace optional parameters of Resize with placeholder for ops13.

* Use common methods to handle different versions.

* Correct flake8 issue.

* Update per comments.

* Add something to trigger CI again.

* Trigger another round of CI.

neginraoof committed:
[ONNX] Fix param names (#50764)

Preserve name of parameters for ONNX.

Looks like output->copyMetadata(input) API is giving the same debugName to the output. So the name of the original input is changed. This update avoid the name change by just copying the type.

neginraoof reviewed (approved):
LGTM. Thanks.

neginraoof commented:
Please rebase this before merge.

fatcat-z committed:
Merge pull request #4 from pytorch/onnx_ms_1

Get latest changes from dev branch.

fatcat-z committed:
Merge branch 'onnx_ms_1' into onnx_ops13_for_split

neginraoof committed:
Merge branch 'onnx_ms_1' into onnx_ops13_for_split

neginraoof committed:
Update test_pytorch_onnx_onnxruntime.py

neginraoof committed:
Update test_pytorch_onnx_onnxruntime.py

fatcat-z committed:
Merge branch 'onnx_ops13_for_split' of https://github.com/fatcat-z/pytorch into onnx_ops13_for_split
